app/uk/gov/hmrc/helptosave/services/EligibilityStatsService.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.services
+
+import cats.instances.int._
+import cats.instances.option._
+import cats.instances.string._
+import cats.syntax.eq._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import uk.gov.hmrc.helptosave.metrics.Metrics
+import uk.gov.hmrc.helptosave.repo.EligibilityStatsStore
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, Logging}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[EligibilityStatsServiceImpl])
+trait EligibilityStatsService {
+
+  def getEligibilityStats()(implicit ec: ExecutionContext): Future[Either[String, String]]
+
+}
+
+@Singleton
+class EligibilityStatsServiceImpl @Inject() (eligibilityReportStore: EligibilityStatsStore,
+                                             metrics:                Metrics)(implicit ninoLogMessageTransformer: LogMessageTransformer)
+  extends EligibilityStatsService with Logging {
+
+  override def getEligibilityStats()(implicit ec: ExecutionContext): Future[Either[String, String]] = {
+    val timerContext = metrics.eligibilityStatsTimer.time()
+    eligibilityReportStore.getEligibilityStats()
+      .map { report ⇒
+        val _ = timerContext.stop()
+
+          def get(reason: Option[Int]): (Int, Int, Int, Int) = {
```

Comment:
any better logic to get the individual metrics ?

Reply:
rather than looking for specific values here, it'd be more general to make the service perhaps use `groupBy` on the list of `EligibilityStats` and group by eligibility reason, then channel. For example, something along the lines of:
```scala
case class X(a: Option[Int], b: Option[String], total: Int)
val l: List[X] =...
val groupedByA: Map[Option[Int], List[X]] =  l.groupBy(_.a)
val groupedByAThenB: List[(String, List[(String, Int)])] =
  groupedByA.map{ case (k,v) ⇒
    k.map(_.toString).getOrElse("Unknown") ->
      v.groupBy(_.b).map(x ⇒ x._1.getOrElse("Unknown") → x._2.map(_.total).sum).toList
  }.toList

```
Then you'd hav to have some logic to sort alphabetically and to insert the totals and to format the table correctly.